### ISSUE TITLE
Can't restore: problem in download_urls(pkgs)

### ISSUE THREAD

**Issue by ManuelaAlmeida** [MangoTheCat/pkgsnap#12]
When I try to restore from a snapshot file I get the following error:

```
Downloading
Error in if (pkgs$Source[i] == "cran") { : 
  missing value where TRUE/FALSE needed
```

**Comment by gaborcsardi** [comment]
Hmmm. Is it possible that you took the snapshot with an older version of `pkgsnap`? Can you please show the output file? Or at least part of it? Thanks.

**Comment by ManuelaAlmeida** [comment]
Thanks for the quick response!

The version of `pkgsnap` was `1.0.0` for both commands. Here is a part of it:

```
Package  Version Source Link
1          R    3.2.3      R   NA
2 assertthat      0.1   cran   NA
3      audio    0.1-5   cran   NA
4      beepr      1.2   cran   NA
5         BH 1.60.0-1   cran   NA
```

**Comment by gaborcsardi** [comment]
`pkgsnap` actually uses csv files, so I am not sure how you got this one..... is it possible that you created the file with something else, or edited it manually, etc?

Anyway, I am pretty sure that `restore` will only work with CSV files.

**Comment by ManuelaAlmeida** [comment]
Oh, this was just a print of `read.csv("snapshot.csv")[1:5,]` (just like you have in README). The first 5 lines of the original file are like this:

```
"Package","Version","Source","Link"
"R","3.2.3","R",NA
"assertthat","0.1","cran",NA
"audio","0.1-5","cran",NA
"beepr","1.2","cran",NA
"BH","1.60.0-1","cran",NA
```

**Comment by gaborcsardi** [comment]
This works fine for me.... can you also write me the output of `traceback()` immediately after the error, and maybe also the whole file? You can also attach it to the bug report. 

And also the output of `devtools::session_info()` if you have devtools installed. Or otherwise sessionInfo()? Thanks.

**Comment by ManuelaAlmeida** [comment]
The `tracebak()`:

```
5: FUN(X[[i]], ...)
4: lapply(seq_len(nrow(pkgs)), function(i) {
       if (pkgs$Source[i] == "cran") {
           cran_file(pkgs$Package[i], pkgs$Version[i])
       }
       else if (pkgs$Source[i] == "bioc") {
           bioc_file(pkgs$Package[i], pkgs$Version[i])
       }
       else if (pkgs$Source[i] == "rforge") {
           rforge_file(pkgs$Package[i], pkgs$Version[i])
       }
       else if (pkgs$Source[i] == "url") {
           pkgs$Link[i]
       }
       else {
           warning("Unknown package source: ", pkgs$repo[i])
           character()
       }
   })
3: download_urls(pkgs)
2: pkg_download(pkgs, dest_dir = tempdir())
1: restore("snapshot.csv")
```

The `devtools::session_info()`:

```
`Session info ------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.2.3 (2015-12-10)
 system   x86_64, mingw32             
 ui       RStudio (0.99.903)          
 language (EN)                        
 collate  English_United States.1252  
 tz       Europe/London               
 date     2016-08-31                  

Packages ----------------------------------------------------------------------------------------------
 package    * version date       source                              
 devtools   * 1.12.0  2016-06-24 CRAN (R 3.2.5)                      
 digest       0.6.10  2016-08-02 CRAN (R 3.2.5)                      
 memoise      1.0.0   2016-01-29 CRAN (R 3.2.5)                      
 pkgsnap    * 1.0.0   2016-08-31 Github (mangothecat/pkgsnap@5b6089c)
 rstudioapi   0.6     2016-06-27 CRAN (R 3.2.5)                      
 withr        1.0.2   2016-06-20 CRAN (R 3.2.5)       
```

[snapshot.zip](https://github.com/MangoTheCat/pkgsnap/files/447557/snapshot.zip)

Thanks for you time!

**Comment by gaborcsardi** [comment]
Thanks! I'll try to look at this in the evening.

**Comment by ManuelaAlmeida** [comment]
Ok, found the problem. I have a local package whose source is NA (so the code fails at `download_urls` when trying to compare `NA` with something -- in this case, right at `cran`).

I also have other packages from MRO for RRE which also have NA as source.

**Comment by gaborcsardi** [comment]
Great! This is a bug then, those packages should be ignored.

I am travelling now but will fix this soon.

**Comment by gaborcsardi** [comment]
Fixed by PR #13
